### PR TITLE
Adding option to use Azurefiles instead of NFS on AKS

### PR DESCRIPTION
**Pull request by Aaquiff**
## Purpose
Currently, NFS is used for persistent storage over other native options due to the fact that it is possible to mount the volume from multiple nodes. This PR adds additional scripts and resources to deploy patterns on AKS using Azurefiles instead of NFS.

### CODE REVIEW

File: advanced/helm/pattern-1/README.md [hunk 1]
```diff
@@ -45,25 +47,34 @@ Git repository. <br>
...
+* Using a Network File System (NFS)
 
-* `sharedDeploymentLocationPath`
-* `analyticsDataLocationPath`
-* `analyticsLocationPath`
+  Create and export unique directories within the NFS server instance for each of the following Kubernetes Persistent Volume
+  resources defined in the `<HELM_HOME>/apim-with-analytics-conf/values.yaml` file:
 
-Grant ownership to `wso2carbon` user and `wso2` group, for each of the previously created directories.
+  * `sharedDeploymentLocationPath`
```
Comment (chirangaalwis) [inline advanced/helm/pattern-1/README.md]: The Persistent Volume configurations that are parameterized differ when compared with what is there in the README [documentation](https://github.com/wso2/kubernetes-apim/blob/0d95d0a3df2816f491b57adb09e8b4a47e3855f2/advanced/helm/pattern-1/apim-with-analytics/values.yaml#L8-L9).

File: advanced/helm/pattern-1/apim-with-analytics/templates/azure/storage-class.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: azurefile
+provisioner: kubernetes.io/azure-file
+mountOptions:
+  - dir_mode=0777
```
Comment (chirangaalwis) [inline advanced/helm/pattern-1/apim-with-analytics/templates/azure/storage-class.yaml]: IMO, the minimum permissions should be 700. WDYT?

File: advanced/helm/pattern-1/apim-with-analytics/templates/azure/azure-rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{{- if (.Values.cloudProvider) and eq .Values.cloudProvider "Azure" }}
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/helm/pattern-1/apim-with-analytics/templates/azure/azure-rbac.yaml]: License header year should be 2019. Change in all relevant, new files.

File: advanced/pattern-2/apim-pub-store-tm/wso2apim-pub-store-tm-volume-claim-azure.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: wso2apim-pattern-2-apim-shared-server-volume-claim
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: azurefile
+  resources:
+    requests:
+      storage: 1Gi
```
Comment (chirangaalwis) [inline advanced/pattern-2/apim-pub-store-tm/wso2apim-pub-store-tm-volume-claim-azure.yaml]: Please add a new line at the EOF.

File: advanced/helm/pattern-1/README.md (hunk 1 above)
Comment (Aaquiff) [reply advanced/helm/pattern-1/README.md]: Changed to,
* sharedDeploymentLocationPath
* sharedTenantsLocationPath

File: advanced/helm/pattern-1/apim-with-analytics/templates/azure/storage-class.yaml (hunk 2 above)
Comment (Aaquiff) [reply advanced/helm/pattern-1/apim-with-analytics/templates/azure/storage-class.yaml]: Changed permission to 0700

File: advanced/helm/pattern-1/apim-with-analytics/templates/azure/azure-rbac.yaml (hunk 3 above)
Comment (Aaquiff) [reply advanced/helm/pattern-1/apim-with-analytics/templates/azure/azure-rbac.yaml]: Fixed

File: advanced/pattern-2/apim-pub-store-tm/wso2apim-pub-store-tm-volume-claim-azure.yaml (hunk 4 above)
Comment (Aaquiff) [reply advanced/pattern-2/apim-pub-store-tm/wso2apim-pub-store-tm-volume-claim-azure.yaml]: Fixed

**Pull request merged**